src/main/scala/com/ibm/couchdb/api/builders/QueryOps.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ibm.couchdb.api.builders
+
+import com.ibm.couchdb.core.Client
+import com.ibm.couchdb.{CouchView, Req}
+import org.http4s.Status
+import upickle.default.Aliases.{R, W}
+import upickle.default._
+
+import scalaz.concurrent.Task
+
+trait QueryOps {
```

Comment:
it could, but I think it gives us more flexibility as a trait ... clients can extend another class but still include the functionality of this trait etc.

however, to reduce the boiler plate of the query methods, we can add an abstract declaration for a client ( `def client: Client`) within the trait that the dependent classes specify/override